ref/address.c

```diff
@@ -1,79 +1,111 @@
 #include <stdint.h>
+#include <string.h>
 
 #include "params.h"
 #include "utils.h"
 
-void addr_to_bytes(unsigned char *bytes, const uint32_t addr[8])
-{
-    int i;
-
-    for (i = 0; i < 8; i++) {
-        ull_to_bytes(bytes + i*4, 4, addr[i]);
-    }
-}
-
+/*
+ * Specify which level of Merkle tree (the "layer") we're working on
+ */
 void set_layer_addr(uint32_t addr[8], uint32_t layer)
 {
-    addr[0] = layer;
+    ((unsigned char *)addr)[SPX_OFFSET_LAYER] = layer;
 }
 
+/*
+ * Specify which Merkle tree within the level (the "tree address") we're working on
+ */
 void set_tree_addr(uint32_t addr[8], uint64_t tree)
 {
 #if (SPX_TREE_HEIGHT * (SPX_D - 1)) > 64
     #error Subtree addressing is currently limited to at most 2^64 trees
 #endif
-    addr[1] = 0;
```

Comment:
Was this one redundant to begin with?

Reply:
The addr structure is always initialized to zero, hence explicitly setting it to zero is redundant (although, as I mentioned elsewhere, we are not time critical here)